gensim/matutils.py

```diff
@@ -382,6 +383,110 @@ def cossim(vec1, vec2):
     return result
 
 
+def isbow(vec):
+    """
+    Checks if vector passed is in bag of words representation or not.
+    """
+    if scipy.sparse.issparse(vec):
```

Comment:
This doesn't sound right -- sparse bag-of-words is not the same format as scipy.sparse.
If scipy.sparse returns True, then so should dense numpy.array... or where's the conceptual line between the formats then?